service/raft/raft_group0_client.cc

```diff
@@ -372,4 +373,44 @@ db::system_keyspace& raft_group0_client::sys_ks() {
     return _sys_ks;
 }
 
+raft_group0_client::query_result_guard::query_result_guard(utils::UUID query_id, raft_group0_client& client)
+    : _query_id{query_id}, _client{&client} {
+    auto it = _client->_results.emplace(_query_id, std::nullopt);
+    if (!it.second) {
+        on_internal_error(logger, "query_result_guard::query_result_guard: creating an entry for the result doesn't succeed");
+    }
+}
+
+raft_group0_client::query_result_guard::query_result_guard(raft_group0_client::query_result_guard&& other)
+    : _query_id{other._query_id}, _client{other._client} {
+    other._client = nullptr;
+}
+
+raft_group0_client::query_result_guard::~query_result_guard() {
+    if (_client != nullptr) {
+        _client->_results.erase(_query_id);
+    }
+}
+
+raft::broadcast_tables::query_result raft_group0_client::query_result_guard::get() const {
```

Comment:
Is this function really logically `const`? We do `move` out of the guarded entry

Reply:
You're right, I removed `const`.